src/ensembl/src/shared/components/in-app-search/InAppSearch.test.tsx

```diff
@@ -0,0 +1,183 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { configureStore } from '@reduxjs/toolkit';
+import { Provider } from 'react-redux';
+import { render, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import apiService from 'src/services/api-service';
+
+import * as inAppSearchSlice from 'src/shared/state/in-app-search/inAppSearchSlice';
+import inAppSearchReducer from 'src/shared/state/in-app-search/inAppSearchSlice';
+
+import InAppSearch, { Props as InAppSearchProps } from './InAppSearch';
+
+import { brca2SearchResults } from './test/response-fixture';
+
+const rootReducer = {
+  inAppSearch: inAppSearchReducer
+};
+
+const getStore = (initialState = {}) => {
+  return configureStore({
+    reducer: rootReducer,
+    devTools: false,
+    preloadedState: initialState
+  });
+};
+
+const defaultProps: InAppSearchProps = {
+  app: 'genomeBrowser',
+  genomeId: 'human',
+  mode: 'interstitial'
+};
+
+describe('<InAppSearch />', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('initial rendering', () => {
+    it('renders correctly before the request', () => {
+      const { container, queryByText } = render(
+        <Provider store={getStore()}>
+          <InAppSearch {...defaultProps} />
+        </Provider>
+      );
+
+      const searchField = container.querySelector('.searchField');
+      expect(searchField).toBeTruthy();
+
+      const label = queryByText('Find a gene in this species');
+      expect(label).toBeTruthy();
+
+      const button = container.querySelector('button');
+      expect(button).toBeTruthy();
+      expect(button?.getAttribute('disabled')).not.toBe(null);
+    });
+
+    it('uses the mode property correctly', () => {
+      const { rerender, queryByTestId } = render(
+        <Provider store={getStore()}>
+          <InAppSearch {...defaultProps} />
+        </Provider>
+      );
+
+      let inAppSearchTop = queryByTestId('in-app search top') as HTMLElement;
+      expect(
+        inAppSearchTop.classList.contains('inAppSearchTopInterstitial')
+      ).toBe(true);
+
+      const sidebarProps = { ...defaultProps, mode: 'sidebar' as const };
+      rerender(
+        <Provider store={getStore()}>
+          s
```

Comment:
I'm confused with this `s` here. Was it added by mistake?

Reply:
Yes. This is my Macbook's horrible keyboard that doesn't always register keys, such as Command. This is the result of an attempt to press Cmd-S to save the file 😠

Reply:
✅